torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1153,6 +1375,21 @@ void ComputeConstant(Node* n, int opset_version) {
         only_rank_available = true;
         rank = rank_0 + rank_1 - 1;
       }
+      if (ConstantValueMap::HasShapeValue(n->input(0)->debugName()) &&
+          ConstantValueMap::HasValue(n->input(1)->debugName())) {
+        auto shape_value =
+            ConstantValueMap::GetShapeValue(n->input(0)->debugName()).value();
+        auto idx_value =
+            ConstantValueMap::GetValue(n->input(1)->debugName()).value();
+        if (idx_value.dim() == 0) {
```

Comment:
Is this corresponding to the special case of gather index being a scalar?

Reply:
Yes, here we consider the case where the gather index is a scalar.

Reply:
Thanks, please add a few comment describing the particular case it's covering, as it was not immediately obvious..

Reply:
Added: 
// Consider the case when Gather index is a scalar.